src/python/pants/core/goals/tailor.py

```diff
@@ -249,6 +250,16 @@ class TailorSubsystem(GoalSubsystem):
     @classmethod
     def register_options(cls, register):
         super().register_options(register)
+        register(
+            "--check",
+            type=bool,
+            default=False,
+            help=(
+                "Do not write changes to disk, only write back what would change. Return code "
+                "0 means there would be no changes, and 1 means that there would be. "
+            ),
+        )
```

Comment:
> As speculated on that ticket, we may eventually want to merge this into lint. But that is non-trivial to do for now, so we focus on the easy win.

Can we open a new ticket that covers a holistic rethink of `tailor`/`fmt`/`update-build-files`, and link it as a TODO somewhere in here? I almost think that calling it out in the help string wouldn't be unwarranted.

Reply:
> Can we open a new ticket that covers a holistic rethink of tailor/fmt/update-build-files

Sure.

> and link it as a TODO somewhere in here?

I can add, but I prefer to not put the TODO in code if that's fine. Based on my experience with helping a new user onboard this week + my upcoming proposal for how to unify `validate` and `lint`, I now think differently about unifying than I did before. I think there's enough discussion to be had to not put in code our proposed plan.

>  I almost think that calling it out in the help string wouldn't be unwarranted.

Benjy has always pushed back when I put speculative future features in help messages like this. Commits to us to something we're not certain we will do